test/unit/master/test_cluster_master.py

```diff
@@ -311,3 +313,140 @@ def test_builds_with_pagination_request(
+            _PAGINATION_OFFSET + 1,
+            _PAGINATION_OFFSET + 5
+        ),
+        offset_and_limit_params=(
+            3, 5,
+            3 + 1,
+            3 + 5
+        ),
+        low_limit=(
+            _PAGINATION_OFFSET, 2,
+            _PAGINATION_OFFSET + 1,
+            _PAGINATION_OFFSET + 2
+        ),
+        max_limit=(
+            _PAGINATION_OFFSET, _PAGINATION_MAX_LIMIT,
+            _PAGINATION_OFFSET + 1,
+            _PAGINATION_OFFSET + _PAGINATION_MAX_LIMIT
+        ),
+        too_high_offset=(
+            1000, _PAGINATION_LIMIT,
+            None,
+            None
+        ),
+    )
+    def test_subjobs_with_pagination_request(
+            self,
+            offset: Optional[int],
+            limit: Optional[int],
+            expected_first_subjob_id: int,
+            expected_last_subjob_id: int,
+            ):
+        build = Build(BuildRequest({}))
+        # Create 20 mock subjobs with ids 1 to 20
+        for subjob_id in range(1, self._NUM_SUBJOBS + 1):
+            subjob_mock = Mock(spec=Subjob)
+            subjob_mock.subjob_id = subjob_id
+            build._all_subjobs_by_id[subjob_id] = subjob_mock
+
+        requested_subjobs = build.all_subjobs(offset, limit)
+
+        id_of_first_subjob = requested_subjobs[0].subjob_id if len(requested_subjobs) else None
+        id_of_last_subjob = requested_subjobs[-1].subjob_id if len(requested_subjobs) else None
+        num_subjobs = len(requested_subjobs)
+
+        self.assertEqual(id_of_first_subjob, expected_first_subjob_id, 'Received the wrong first subjob from request')
+        self.assertEqual(id_of_last_subjob, expected_last_subjob_id, 'Received the wrong last subjob from request')
+        if offset is not None and limit is not None:
+            self.assertLessEqual(num_subjobs, self._PAGINATION_MAX_LIMIT, 'Received too many subjobs from request')
+
+
+
+    @genty_dataset(
+        # No params simulates a v1 request
+        no_params=(
+            None, None,
+            1,
+            0 + _NUM_ATOMS
+        ),
+        # Params simulate a v2 request
+        offset_param=(
+            3, _PAGINATION_LIMIT,
+            3 + 1,
+            3 + _PAGINATION_LIMIT
+        ),
+        limit_param=(
+            _PAGINATION_OFFSET, 5,
+            _PAGINATION_OFFSET + 1,
+            _PAGINATION_OFFSET + 5
+        ),
+        offset_and_limit_params=(
+            3, 5,
+            3 + 1,
+            3 + 5
+        ),
+        low_limit=(
+            _PAGINATION_OFFSET, 2,
+            _PAGINATION_OFFSET + 1,
+            _PAGINATION_OFFSET + 2
+        ),
+        max_limit=(
+            _PAGINATION_OFFSET, _PAGINATION_MAX_LIMIT,
+            _PAGINATION_OFFSET + 1,
+            _PAGINATION_OFFSET + _PAGINATION_MAX_LIMIT
+        ),
+        too_high_offset=(
+            1000, _PAGINATION_LIMIT,
+            None,
+            None
+        ),
+    )
+    def test_subjobs_with_pagination_request(
```

Comment:
Did you mean to name this something else? I'm guessing only one of these tests runs.